### PR TITLE
First commit

### PR DESCRIPTION
**Pull request by gigbat**

### CODE REVIEW

File: src/main/java/core/basesyntax/Main.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        Handler consoleHandlerBet = new BetConsoleHandler();
+        consoleHandlerBet.handler();
+        BetDao betDao = new BetDaoImpl();
+        List<Bet> listBet = betDao.getAll();
+
+        System.out.println("Please, enter 'value' and 'risk' for you bet.");
+        Handler consoleHandlerFruits = new FruitConsoleHandler();
+        consoleHandlerFruits.handler();
+        FruitDao fruitDao = new FruitDaoImpl();
+        List<Fruit> listFruit = fruitDao.getAll();
```
Comment (ivlytvynets) [inline src/main/java/core/basesyntax/Main.java]:
```suggestion
         ConsoleHandler handler = new ConsoleHandler();
        System.out.println("Enter value and risk fpr your bet");
        handler.handle();
```

**Review by ivlytvynets — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Main.java (hunk 1 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/Main.java]: Also for class Fruit, I added for the call handler

File: src/main/java/core/basesyntax/Main.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import core.basesyntax.model.Fruit;
+import java.util.List;
+
+public class Main {
+    public static void main(String[] args) {
+        System.out.println("Please, enter 'value' and 'risk' for your bets.");
+        Handler consoleHandlerBet = new BetConsoleHandler();
+        consoleHandlerBet.handler();
+
+        System.out.println("Please, enter 'value' and 'risk' for your fruits.");
```
Comment (vlykhvar) [inline src/main/java/core/basesyntax/Main.java]: you don't have fields 'value' and 'risk' for fruits. should make message understandable  for customers

File: src/main/java/core/basesyntax/handler/FruitConsoleHandler.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+                return;
+            }
+            try {
+                String[] parseString = input.split(" ");
+                String name = parseString[0];
+                int quantity = Integer.parseInt(parseString[1]);
+                fruit = new Fruit(name, quantity);
+                System.out.println(fruit.toString());
+            } catch (Exception e) {
+                throw new RuntimeException("Enter only digital number.");
```
Comment (vlykhvar) [inline src/main/java/core/basesyntax/handler/FruitConsoleHandler.java]: if I wrote null name and write right quantity that's would not be great explain "Enter only digital number."

**Review by ivlytvynets — APPROVED** [review]

**Review by vlykhvar — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Main.java (hunk 2 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/Main.java]: removed it, i released controller only for bet

File: src/main/java/core/basesyntax/handler/FruitConsoleHandler.java (hunk 3 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/handler/FruitConsoleHandler.java]: exactly, thanks

**Review by vlykhvar — APPROVED** [review]

File: pom.xml [hunk 4]
```diff
@@ -25,6 +25,12 @@
             <version>4.12</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <version>RELEASE</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (KseniiaMakarova) [inline pom.xml]: do we need this?

File: src/main/java/core/basesyntax/dao/FruitDao.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.model.Fruit;
+import java.util.List;
+
+public interface FruitDao {
+    void add(Fruit bet);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/dao/FruitDao.java]: bet?

File: src/main/java/core/basesyntax/handler/BetConsoleHandler.java [hunk 6]
```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.handler;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.BetDaoImpl;
+import core.basesyntax.model.Bet;
+import java.util.Scanner;
+
+public class BetConsoleHandler implements Handler {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/handler/BetConsoleHandler.java]: let’s have only one handler that will read info about both entities from the console
example of user input `200 0.6 banana 20`

File: src/main/java/core/basesyntax/handler/BetConsoleHandler.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    BetDao betDao = new BetDaoImpl();
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/handler/BetConsoleHandler.java]: what is missing?

File: src/main/java/core/basesyntax/handler/BetConsoleHandler.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Override
+    public void handler() {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/handler/BetConsoleHandler.java]:
```suggestion
    public void handle() {
```

File: src/main/java/core/basesyntax/handler/BetConsoleHandler.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+            if (input.equalsIgnoreCase("q")) {
+                return;
+            }
+            try {
+                String[] parseString = input.split(" ");
+                int value = Integer.parseInt(parseString[0]);
+                double risk = Integer.parseInt(parseString[1]);
+                bet = new Bet(value, risk);
+                System.out.println(bet.toString());
+            } catch (Exception e) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/handler/BetConsoleHandler.java]: let's narrow this catch

File: src/main/java/core/basesyntax/handler/BetConsoleHandler.java [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+                throw new RuntimeException("Enter only digital number.");
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/handler/BetConsoleHandler.java]: use System.out.println() to give user an opportunity to enter correct values

File: src/main/java/core/basesyntax/model/Bet.java [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+        if (value < 0) {
+            throw new IllegalArgumentException("Value is negative");
+        }
+    }
+
+    private void checkNegativeRisk(double risk) {
+        if (risk < 0) {
+            throw new IllegalArgumentException("Value is negative");
+        }
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/model/Bet.java]: let's check these in the handler

File: src/main/java/core/basesyntax/model/Fruit.java [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    public void setQuantity(int quantity) {
+        checkForNegativeQuantity(quantity);
+        this.quantity = quantity;
+    }
+
+    private void checkForNegativeQuantity(int quantity) {
+        if (quantity < 0) {
+            throw new RuntimeException("Value is negative");
+        }
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/model/Fruit.java]: same, as well as null check

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 4 above)
Comment (gigbat) [reply pom.xml]: done

File: src/main/java/core/basesyntax/dao/FruitDao.java (hunk 5 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/dao/FruitDao.java]: renamed

File: src/main/java/core/basesyntax/handler/BetConsoleHandler.java (hunk 6 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/handler/BetConsoleHandler.java]: done

Comment (gigbat) [reply src/main/java/core/basesyntax/handler/BetConsoleHandler.java]: removed FruitHandler

File: src/main/java/core/basesyntax/handler/BetConsoleHandler.java (hunk 7 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/handler/BetConsoleHandler.java]: private)

File: src/main/java/core/basesyntax/handler/BetConsoleHandler.java (hunk 8 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/handler/BetConsoleHandler.java]: done

File: src/main/java/core/basesyntax/handler/BetConsoleHandler.java (hunk 9 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/handler/BetConsoleHandler.java]: done

File: src/main/java/core/basesyntax/handler/BetConsoleHandler.java (hunk 10 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/handler/BetConsoleHandler.java]: done, i hope that i understood you

File: src/main/java/core/basesyntax/model/Bet.java (hunk 11 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/model/Bet.java]: done

File: src/main/java/core/basesyntax/model/Fruit.java (hunk 12 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/model/Fruit.java]: I removed FruitHandler => i removed these methods into the class Fruit

File: src/main/java/core/basesyntax/Main.java [hunk 13]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import core.basesyntax.handler.ConsoleHandler;
+import core.basesyntax.handler.Handler;
+
+public class Main {
+    public static void main(String[] args) {
+        System.out.println("Please, enter 'value' and 'risk' for your bets.");
+        Handler consoleHandlerBet = new ConsoleHandler();
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Main.java]:
```suggestion
        Handler consoleHandler = new ConsoleHandler();
```

File: src/main/java/core/basesyntax/handler/ConsoleHandler.java [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+            if (input.equalsIgnoreCase("q")) {
+                return;
+            }
+            try {
+                String[] parseString = input.split(" ");
+                int value = Integer.parseInt(parseString[0]);
+                double risk = Integer.parseInt(parseString[1]);
+
+                checkNegativeValue(value);
+                checkNegativeRisk(risk);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/handler/ConsoleHandler.java]: throwing and then catching RuntimeExceptions is not a good practice
let's remove private methods, combine the two checks in one `if` and sout some message, then call `continue`

File: src/main/java/core/basesyntax/handler/ConsoleHandler.java [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+
+                bet = new Bet(value, risk);
+            } catch (IllegalArgumentException e) {
+                System.out.println("Your data is - " + bet.toString());
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/handler/ConsoleHandler.java]: what happens if bet is null at this point?

File: src/main/java/core/basesyntax/handler/ConsoleHandler.java [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+            try {
+                String[] parseString = input.split(" ");
+                int value = Integer.parseInt(parseString[0]);
+                double risk = Integer.parseInt(parseString[1]);
+
+                checkNegativeValue(value);
+                checkNegativeRisk(risk);
+
+                bet = new Bet(value, risk);
+            } catch (IllegalArgumentException e) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/handler/ConsoleHandler.java]: let's catch NumberFormatException or ArrayIndexOutOfBoundException

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/handler/ConsoleHandler.java [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+            if (input.equalsIgnoreCase("q")) {
+                return;
+            }
+            try {
+                String[] parseString = input.split(" ");
+                int value = Integer.parseInt(parseString[0]);
+                double risk = Integer.parseInt(parseString[1]);
+
+                if (risk < 0 || value < 0) {
+                    System.out.println("Value is negative");
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/handler/ConsoleHandler.java:27]: should we skip the rest of code in the current iteration?

File: src/main/java/core/basesyntax/handler/ConsoleHandler.java [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+                }
+                bet = new Bet(value, risk);
+            } catch (NumberFormatException | ArrayIndexOutOfBoundsException e) {
+                System.out.println("Your data is - " + (bet != null ? bet.toString() : ""));
+                throw new RuntimeException("Input isn't correct.");
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/handler/ConsoleHandler.java]:
```suggestion
                System.out.println("Enter correct values. Your data is - " + (bet != null ? bet.toString() : ""));
```
can we give the opportunity to the user to fix the mistake and enter correct values?

File: src/main/java/core/basesyntax/model/Fruit.java [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    @Override
+    public String toString() {
+        return "Bet = [" + name + " " + quantity + "]";
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/model/Fruit.java]: why Bet?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/handler/ConsoleHandler.java (hunk 17 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/handler/ConsoleHandler.java:27]: Above you wrote to me that i continue my implementation. I'm a little confused already :)

Comment (gigbat) [reply src/main/java/core/basesyntax/handler/ConsoleHandler.java:27]: Okey i throw RunTimeException

File: src/main/java/core/basesyntax/handler/ConsoleHandler.java (hunk 18 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/handler/ConsoleHandler.java]: Yes, in the next iteration

File: src/main/java/core/basesyntax/model/Fruit.java (hunk 19 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/model/Fruit.java]: I forgot, sorry

File: src/main/java/core/basesyntax/handler/ConsoleHandler.java [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+                return;
+            }
+            try {
+                String[] parseString = input.split(" ");
+                int value = Integer.parseInt(parseString[0]);
+                double risk = Integer.parseInt(parseString[1]);
+
+                if (risk < 0 || value < 0) {
+                    throw new RuntimeException("Value is negative");
+                }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/handler/ConsoleHandler.java]:
```suggestion
                if (risk < 0 || value < 0) {
                    System.out.println("Value is negative");
                    continue;
                }
```

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/core/basesyntax/handler/ConsoleHandler.java (hunk 20 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/handler/ConsoleHandler.java:28]: Can I don't to add the keyword continue?

Comment (gigbat) [reply src/main/java/core/basesyntax/handler/ConsoleHandler.java]: Done

**Pull request merged**